File: test-suite/src/test/java/io/micronaut/docs/metrics/annotation/MethodNameTagger.java
package io.micronaut.docs.metrics.annotation;

// tag::imports[]
import io.micrometer.core.instrument.Tag;
import io.micronaut.aop.MethodInvocationContext;
import io.micronaut.configuration.metrics.aggregator.AbstractMethodTagger;
import jakarta.inject.Singleton;

import java.util.Collections;
import java.util.List;
// end::imports[]

// tag::class[]
@Singleton
public class MethodNameTagger extends AbstractMethodTagger {
    @Override
    public List<Tag> buildTags(MethodInvocationContext<Object, Object> context) {
        return Collections.singletonList(Tag.of("method", context.getMethodName()));
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/metrics/annotation/MethodNameTagger.py
# tag::imports[]
from io.micrometer.core.instrument import Tag
from jakarta.inject import Singleton
from micronaut.aop import MethodInvocationContext
from micronaut.configuration.metrics.aggregator import AbstractMethodTagger
# end::imports[]


# tag::class[]
@Singleton
class MethodNameTagger(AbstractMethodTagger):
    def buildTags(self, context: MethodInvocationContext) -> list[Tag]:
        return [Tag.of("method", context.getMethodName())]
# end::class[]
